Sources/TelemetryDeck/Presets/TelemetryDeck+Purchases.swift

```diff
@@ -19,230 +19,285 @@ extension TelemetryDeck {
-            countryCode = transaction.storefront.countryCode
+            countryCode = self.storefront.countryCode
         } else {
             #if os(visionOS)
             countryCode = "US"
             #else
-            countryCode = transaction.storefrontCountryCode
+            countryCode = self.storefrontCountryCode
             #endif
         }
 
         var purchaseParameters: [String: String] = [
-            "TelemetryDeck.Purchase.type": transaction.subscriptionGroupID != nil ? "subscription" : "one-time-purchase",
+            "TelemetryDeck.Purchase.type": self.subscriptionGroupID != nil ? "subscription" : "one-time-purchase",
             "TelemetryDeck.Purchase.countryCode": countryCode,
+            "TelemetryDeck.Purchase.productID": self.productID,
         ]
 
         if #available(iOS 16, macOS 13, tvOS 16, watchOS 9, *) {
-            if let currencyCode = transaction.currency?.identifier {
+            if let currencyCode = self.currency?.identifier {
                 purchaseParameters["TelemetryDeck.Purchase.currencyCode"] = currencyCode
             }
         } else {
-            if let currencyCode = transaction.currencyCode {
+            if let currencyCode = self.currencyCode {
                 purchaseParameters["TelemetryDeck.Purchase.currencyCode"] = currencyCode
             }
         }
 
-        self.internalSignal(
-            "TelemetryDeck.Purchase.completed",
-            parameters: purchaseParameters.merging(parameters) { $1 },
-            floatValue: priceValueInUSD,
-            customUserID: customUserID
-        )
+        return purchaseParameters
+    }
+
+    func priceInUSD() -> Double {
+        let priceValueInNativeCurrency = NSDecimalNumber(decimal: self.price ?? Decimal()).doubleValue
+        let priceValueInUSD: Double
+
+        if #available(iOS 16, macOS 13, tvOS 16, watchOS 9, *) {
+            if self.currency?.identifier == "USD" {
+                priceValueInUSD = priceValueInNativeCurrency
+            } else if
+                let currencyCode = self.currency?.identifier,
+                let oneUSDExchangeRate = Self.currencyCodeToOneUSDExchangeRate[currencyCode]
+            {
+                priceValueInUSD = priceValueInNativeCurrency / oneUSDExchangeRate
+            } else {
+                priceValueInUSD = 0
+            }
+        } else {
+            if self.currencyCode == "USD" {
+                priceValueInUSD = priceValueInNativeCurrency
+            } else if
+                let currencyCode = self.currencyCode,
+                let oneUSDExchangeRate = Self.currencyCodeToOneUSDExchangeRate[currencyCode]
+            {
+                priceValueInUSD = priceValueInNativeCurrency / oneUSDExchangeRate
+            } else {
+                priceValueInUSD = 0
+            }
+        }
+
+        return priceValueInUSD
     }
 
     private static let currencyCodeToOneUSDExchangeRate: [String: Double] = [
```

Comment:
@Jeehut @winsmith are we using a specific source for the currency rates? How often do we plan to have these updated?